### PR TITLE
*: Use glide in Makefile

### PR DESCRIPTION
**Pull request by ericchiang**
When building from source, don't recommend using "go get". Instead
clone and populate the vendor directory using glide.

This updates the perceived issue with #189

The "go get" was removed because it downloads a lot of packages but
not enough to actually build torus. See comment [here](
https://github.com/coreos/torus/pull/190#issuecomment-223345034).

Since this introduces another binary, I've moved the default build to
use a "bin" directory.

cc @barakmich

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,19 +3,22 @@ ifeq ($(origin VERSION), undefined)
 endif
 REPOPATH = github.com/coreos/torus
 
-build:
-	go build -ldflags "-X $(REPOPATH).Version=$(VERSION)" ./cmd/torusd 
-	go build -ldflags "-X $(REPOPATH).Version=$(VERSION)" ./cmd/torusctl 
-	go build -ldflags "-X $(REPOPATH).Version=$(VERSION)" ./cmd/torusblk 
+# Use the go tool to determine OS and architecture.
+OS_AND_ARCH != go version | awk '{ print $$(NF)}' | tr "/" "-"
```
Comment (ericchiang) [inline Makefile]: For some reason this doesn't work on travis, investigating.

**Comment by barakmich** [comment]
Glide now doesn't suck -- rebase, and it's all you @ericchiang

**Comment by ericchiang** [comment]
TL;DR - Travis' make doesn't understand "FOO != ./cmd" syntax.

Anyway looks like I found some bugs by turning on the race detector :)

**Comment by ericchiang** [comment]
Going to turn off the race detector for now then turn it back on with a fix for #201.

File: Documentation/getting-started.md [hunk 2]
```diff
@@ -68,8 +59,8 @@ will tell you more, check the docs, or feel free to ask in IRC.
 ### 3) Run some storage nodes
 #### Running manually
 ```
-./torusd --etcd 127.0.0.1:2379 --peer-address http://127.0.0.1:40000 --data-dir /tmp/torus1 --size 20GiB
-./torusd --etcd 127.0.0.1:2379 --peer-address http://127.0.0.1:40001 --data-dir /tmp/torus2 --size 20GiB
+torusd --etcd 127.0.0.1:2379 --peer-address http://127.0.0.1:40000 --data-dir /tmp/torus1 --size 20GiB
```
Comment (barakmich) [inline Documentation/getting-started.md]: it also installs them? or is this ./bin/

Comment (ericchiang) [inline Documentation/getting-started.md]: It doesn't install them. These docs mix "./torusd" and "torusd" (lots of "torusctl" and "torusblk" commands) but the latter case was more common so I standardized it on that.

Would be happy to switch to prefixing all command with ./bin.

Comment (barakmich) [inline Documentation/getting-started.md]: Make it so. :+1:

Comment (ericchiang) [inline Documentation/getting-started.md]: done

**Comment by barakmich** [comment]
LGTM
